### PR TITLE
feat(platform/gitea): support `automergeStrategy`

### LINKED ISSUE #10868: Implement the explicit `automergeStrategy` for the Gitea platform.
**What would you like Renovate to be able to do?**

The `automergeStrategy` configuration item introduced in #10627 should be implemented for the Gitea platform, allowing users to configure the explicit merge strategy used by Renovate when automerging pull requests

**Did you already have any implementation ideas?**

The optional `MergeStrategy` argument can be provided to the Gitea `mergePr` implementation. The current implementation uses the merge strategy configured on the repository at the time it is internally constructed, with preferences based on which features are allowed. The Renovate-specific strings for automergeStrategy can be mapped to Gitea-specific strings when submitting the merge request to Gitea or aborted if the configured strategy is not listed as available.

If #10830 has been implemented, the arguments to `mergePr` may be different than what has been listed here. We should also remember to clean up the missing feature notice in `lib/platform/gitea/index.md` and update the description in `docs/usage/configuration-options.md` if necessary.

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- support `automergeStrategy`

## Context
- closes #10868

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitea/utils.ts [hunk 1]
```diff
@@ -39,7 +40,6 @@ export function getRepoUrl(
     if (!url) {
       throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
     }
-    url.protocol = url.protocol?.slice(0, -1) ?? 'https';
```
Comment (viceice) [inline lib/modules/platform/gitea/utils.ts:42]: Tried to cover altenate branch but it seems to never happen, so removed

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -192,7 +192,7 @@ Acceptable values are:
 Not all platforms support all pull request merge strategies.
 In cases where a merge strategy is not supported by the platform, Renovate will hold off on merging instead of silently merging in a way you didn't wish for.
 
-The only platform that supports `automergeStrategy` is Bitbucket Cloud.
+The only platforms that support `automergeStrategy` is Bitbucket Cloud and Gitea.
```
Comment (viceice) [inline docs/usage/configuration-options.md:195]:
```suggestion
The only platforms that support `automergeStrategy` are Bitbucket Cloud and Gitea.
```

**Review by HonkingGoose — DISMISSED** [review]
Docs look good to me.

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -192,7 +192,7 @@ Acceptable values are:
 Not all platforms support all pull request merge strategies.
 In cases where a merge strategy is not supported by the platform, Renovate will hold off on merging instead of silently merging in a way you didn't wish for.
 
-The only platform that supports `automergeStrategy` is Bitbucket Cloud.
+The only platforms that support `automergeStrategy` are Bitbucket Cloud and Gitea.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:195]: I think this could removed now that we expose supportedPlatforms

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -192,7 +192,7 @@ Acceptable values are:
...
```
Comment (viceice) [inline docs/usage/configuration-options.md:196]:
```suggestion
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**